src/core/settings/qgssettingsregistrycore.cpp

```diff
@@ -151,7 +151,7 @@ void QgsSettingsRegistryCore::migrateOldSettings()
   pal::Pal::settingsRenderingLabelCandidatesLimitPolygons->copyValueFromKey( QStringLiteral( "core/rendering/label_candidates_limit_polygons" ), true );
 
   // migrate only one way for map scales
-  if ( !settingsMapScales->exists() )
+  if ( !settingsMapScales->exists() && QgsSettings().contains( QStringLiteral( "Map/scales" ) ) )
```

Comment:
// handle bad migration (TODO remove in 3.32 or later)
  if ( settingsMapScales->value() == QString( "" ) )
    settingsMapScales->remove();

Adding this should fix former issues